src/libraries/System.IO.Pipes/tests/NamedPipeTests/NamedPipeTest.MessageMode.Windows.cs

```diff
@@ -0,0 +1,106 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+using System.IO.Tests;
+using System.Linq;
+using System.Runtime.InteropServices;
+using System.Threading;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace System.IO.Pipes.Tests
+{
+    // Support for PipeAccessRights and setting ReadMode to Message is only supported on Windows
+    public class NamedPipeTest_MessageMode_Windows
+    {
+        private const PipeAccessRights MinimumMessageAccessRights = PipeAccessRights.ReadData | PipeAccessRights.WriteAttributes;
+
+        private static NamedPipeClientStream CreateClientStream(string pipeName, PipeOptions options) =>
+            new NamedPipeClientStream(".", pipeName, MinimumMessageAccessRights, options, Security.Principal.TokenImpersonationLevel.None, HandleInheritability.None);
+
+        [Theory]
+        [InlineData(PipeDirection.Out, PipeOptions.None)]
+        [InlineData(PipeDirection.InOut, PipeOptions.Asynchronous)]
+        public async Task Client_DetectsMessageCompleted(PipeDirection serverDirection, PipeOptions options)
+        {
+            string pipeName = PipeStreamConformanceTests.GetUniquePipeName();
+
+            using NamedPipeServerStream server = new NamedPipeServerStream(pipeName, serverDirection, 1, PipeTransmissionMode.Message, options);
+            using NamedPipeClientStream client = CreateClientStream(pipeName, options);
+
+            Task.WaitAll(server.WaitForConnectionAsync(), client.ConnectAsync());
+            client.ReadMode = PipeTransmissionMode.Message;
+
+            ValueTask serverWrite = server.WriteAsync(new byte[] { 1, 2, 3, 4, 5 });
```

Comment:
Is there a reason why the `ValueTask` is being created here if it's not being awaited? I'm failing to see if such early/non-awaited call is relevant to the new constructor.

Reply:
It is awaited before existing the test. I need to call the non-blocking `WriteAsync` so that the test can proceed and the client can start consuming the data; otherwise, `Write` blocks while waiting for a client to consume the data.

I could probably get away with discarding the `ValueTask`, but it seemed cleaner to capture it and await it before letting the test finish.